main/ui/src/main/java/org/cryptomator/ui/fxapp/FxApplication.java

```diff
@@ -192,4 +198,21 @@ private void applyDarkTheme() {
 		});
 	}
 
+
+
+	private void checkAutolock(Vault vault,  Optional<Stage> owner){
+		if (vault.getVaultSettings().lockAfterTime().get()){
+			LOG.info("Locking after {} minutes.", vault.getVaultSettings().lockTimeInMinutes().get());
+			new java.util.Timer().schedule(
```

Comment:
Don't create a new timer, inject a `ScheduledExecutorService` instead. Its API will allow roughly the same.